openwrt-openwisp-monitoring/files/monitoring.agent

```diff
@@ -267,6 +285,11 @@ main() {
 		MONITORED_INTERFACES=$(echo "$MONITORED_INTERFACES" | tr -d '"')
 		save_data
 	elif [ "$MODE" = "send" ]; then
+		if [ -z "$UUID" ] || [ -z "$KEY" ]; then
+			until register; do
+				sleep "$REGISTRATION_INTERVAL"
+			done
+		fi
```

Comment:
Waiting for uuid and key to be set in `send` mode only because collect mode does not depend on uuid and key.